numpy/core/src/umath/umathmodule.c

```diff
@@ -101,40 +101,13 @@ ufunc_frompyfunc(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *NPY_UNUS
         PyErr_SetString(PyExc_TypeError, "function must be callable");
         return NULL;
     }
-    if (nin + nout > NPY_MAXARGS) {
-        PyErr_Format(PyExc_ValueError,
-                     "Cannot construct a ufunc with more than %d operands "
-                     "(requested number were: inputs = %d and outputs = %d)",
-                     NPY_MAXARGS, nin, nout);
-        return NULL;
-    }
-    self = PyArray_malloc(sizeof(PyUFuncObject));
+    self = PyUFunc_FromFuncAndData(pyfunc_functions, /*data*/ NULL,
+                    /* types */ NULL, /* ntypes */ 1, nin, nout, PyUFunc_None,
+                    /* name */ NULL, /* doc */ NULL, /* unused */ 0);
```

Comment:
Instead of passing in `NULL` here, could you call this function after you have values for these arguments?